packages/plugins/persisted-queries/src/plugin.ts

```diff
@@ -0,0 +1,96 @@
+/* eslint-disable no-console */
+import { Plugin, DefaultContext } from '@envelop/core';
+import { DocumentNode, parse, Source, GraphQLError } from 'graphql';
+
+const contextProperty = 'documentId';
+
+export type PersistedQueriesStoreList = Map<string, { [key: string]: string }>;
+
+export interface PersistedQueriesStore {
+  get(): PersistedQueriesStoreList;
+}
+
+interface PluginContext {
+  [contextProperty]?: string;
+}
+
+export type UsePersistedQueriesOptions = {
```

Comment:
I wonder if these can be moved into the existing `persisted-operations` plugin? 
And then `pickSingleList` might be flag for the store?

Reply:
Yes, but then you need to be prescriptive of store shape (key/value Map).
So your signature changes.